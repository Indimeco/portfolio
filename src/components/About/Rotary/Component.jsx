import React, { useState, useEffect } from 'react';
import { CSSTransition } from 'react-transition-group';
import {
	RotaryView,
	RotaryWrapper,
	RotaryText,
	RotaryWord,
} from './Component.style';
import { rotary } from '../../../content';
import { WATERFALL_2 } from '../../UI/base';
import { FadeOnScroll } from '../../UI/snippets';
const { words, text } = rotary;

export const Rotary = () => {
	const [word, setWord] = useState(words[0]);
	const [showWord, setShowWord] = useState(false);

	useEffect(() => {
		setShowWord(true);
	}, []);

	function hideWord() {
		setShowWord(false);
	}

	function changeWord() {
		const currentIndex = words.findIndex(i => i === word);
		const nextIndex = currentIndex + 1 < words.length ? currentIndex + 1 : 0;

		setWord(words[nextIndex]);
		setShowWord(true);
	}

	return (
		<RotaryView>
			<RotaryWrapper data-test="rotary">
				<FadeOnScroll>
					<RotaryText>{text}</RotaryText>
					<CSSTransition
						in={showWord}
						classNames="fade"
						timeout={WATERFALL_2}
						onEntered={hideWord}
						onExited={changeWord}
					>
						<RotaryWord>{word}</RotaryWord>
					</CSSTransition>
				</FadeOnScroll>
			</RotaryWrapper>
		</RotaryView>
	);
};

export default Rotary;
