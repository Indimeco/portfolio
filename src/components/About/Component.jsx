import React from 'react';
import Rotary from './Rotary';
import Log from './Log';

export const About = () => {
	return (
		<div data-test="about">
			<Rotary />
			<Log />
		</div>
	);
};

export default About;
