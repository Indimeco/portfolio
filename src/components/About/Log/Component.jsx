import React from 'react';
import { LogView, Entry, EntryText, EntryIcon, Blink } from './Component.style';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { about } from '../../../content';
import { FadeOnScroll } from '../../UI/snippets';

const BlinkRunner = () => (
	<EntryText>
		<Blink>_</Blink>
	</EntryText>
);

const Terminal = () => <EntryText>> </EntryText>;

const Log = () => {
	return (
		<LogView data-test="log">
			<div>
				{about.map((entry, index) => (
					<FadeOnScroll key={`entry${index}`}>
						<Entry data-test="log-entry" altStyle={index % 2 !== 0}>
							<Terminal />
							{entry.map((item, itemIndex) =>
								typeof item === 'string' ? (
									<EntryText key={`entry-text${index}-${itemIndex}`}>
										{item}
									</EntryText>
								) : (
									<EntryIcon
										key={`entry-icon${index}-${itemIndex}`}
										data-test="log-entry-icon"
									>
										<FontAwesomeIcon icon={item}></FontAwesomeIcon>
									</EntryIcon>
								),
							)}
							<BlinkRunner />
						</Entry>
					</FadeOnScroll>
				))}
			</div>
		</LogView>
	);
};

export default Log;
