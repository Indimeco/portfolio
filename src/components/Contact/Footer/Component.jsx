import React from 'react';

import { portfolioSourceCode } from '../../../content';

import { FooterWrapper, FooterContent } from './Component.style';

export const Footer = () => {
	return (
		<FooterWrapper data-testid="footer">
			<FooterContent href={portfolioSourceCode} target="_blank">
				View this page's source code
			</FooterContent>
		</FooterWrapper>
	);
};

export default Footer;
