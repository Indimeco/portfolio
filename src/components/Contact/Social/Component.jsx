import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { social } from '../../../content';
import { socialId } from '../../../content/social';

import { SocialWrapper, SocialLink } from './Component.style';

export const Social = () => (
	<SocialWrapper data-testid="social" id={socialId}>
		{social.map(({ href, icon, name }) => (
			<SocialLink key={`social${href}`} href={href} aria-label={name} target="_blank">
				<FontAwesomeIcon icon={icon} />
			</SocialLink>
		))}
	</SocialWrapper>
);

export default Social;
