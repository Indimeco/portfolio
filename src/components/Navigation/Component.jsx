import React from 'react';
import { NavigationWrapper } from './Component.style';


const Navigation = () => <NavigationWrapper>Some text</NavigationWrapper>

export default Navigation;