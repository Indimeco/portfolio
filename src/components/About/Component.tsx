import React from 'react';

import { aboutId } from '../../content/about';

import Log from './Log';
import { AboutWrapper } from './Component.style';

export const About: React.FunctionComponent = () => (
	<AboutWrapper data-testid="about" id={aboutId}>
		<Log />
	</AboutWrapper>
);

export default About;
