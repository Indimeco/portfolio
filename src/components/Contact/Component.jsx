import React from 'react';
import Social from './Social';
import Footer from './Footer';
import {
	ContactView,
	ContactWrapper,
	ContactHeading,
} from './Component.style.js';

export const Contact = () => {
	return (
		<ContactView>
			<ContactWrapper>
				<ContactHeading as="h2">Find me online</ContactHeading>
				<Social />
			</ContactWrapper>
			<Footer />
		</ContactView>
	);
};

export default Contact;
