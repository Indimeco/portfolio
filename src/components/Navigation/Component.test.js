import React from 'react';
import renderer from 'react-test-renderer';
import Component from './Component';
import { ThemeProvider } from 'styled-components';
import { theme } from '../Layout/theme';

describe('Navigation', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<ThemeProvider theme={theme}>
					<Component />
				</ThemeProvider>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
