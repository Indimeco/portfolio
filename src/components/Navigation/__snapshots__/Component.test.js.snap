// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div
  className="sc-bdVaJa kroqVh"
>
  Some text
</div>
`;
